### PR TITLE
Map some datapack related functions

### PR DESCRIPTION
**Pull request by joshyrobot**

### CODE REVIEW

File: data/net/minecraft/server/packs/repository/Pack.mapping [hunk 1]
```diff
@@ -1,4 +1,39 @@
 CLASS net/minecraft/server/packs/repository/Pack
+	METHOD <init> (Ljava/lang/String;Lnet/minecraft/network/chat/Component;ZLjava/util/function/Supplier;Lnet/minecraft/server/packs/metadata/pack/PackMetadataSection;Lnet/minecraft/server/packs/PackType;Lnet/minecraft/server/packs/repository/Pack$Position;Lnet/minecraft/server/packs/repository/PackSource;)V
+		ARG 1 id
+		ARG 2 component
```
Comment (sciwhiz12) [inline data/net/minecraft/server/packs/repository/Pack.mapping]: This seems to be assigned to the `title` field.

File: data/net/minecraft/server/packs/repository/Pack.mapping [hunk 2]
```diff
@@ -1,4 +1,39 @@
...
+		ARG 4 supplier
+		ARG 5 metadata
+		ARG 6 type
+		ARG 7 defaultPosition
+		ARG 8 packSource
+	METHOD <init> (Ljava/lang/String;ZLjava/util/function/Supplier;Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/server/packs/repository/PackCompatibility;Lnet/minecraft/server/packs/repository/Pack$Position;ZLnet/minecraft/server/packs/repository/PackSource;)V
+		ARG 1 id
+		ARG 2 required
+		ARG 3 supplier
+		ARG 4 component
```
Comment (sciwhiz12) [inline data/net/minecraft/server/packs/repository/Pack.mapping]: This seems to be assigned to the `title` field.

File: data/net/minecraft/server/packs/repository/Pack.mapping [hunk 3]
```diff
@@ -1,4 +1,39 @@
...
 		ARG 3 factory
+		ARG 4 defaultPosition
+		ARG 5 packSource
+	METHOD getChatLink (Z)Lnet/minecraft/network/chat/Component;
+		ARG 1 green
+			COMMENT used to indicate either a successful operation or datapack enabled status
+	CLASS PackConstructor
+		METHOD create (Ljava/lang/String;Lnet/minecraft/network/chat/Component;ZLjava/util/function/Supplier;Lnet/minecraft/server/packs/metadata/pack/PackMetadataSection;Lnet/minecraft/server/packs/repository/Pack$Position;Lnet/minecraft/server/packs/repository/PackSource;)Lnet/minecraft/server/packs/repository/Pack;
+			ARG 1 id
+			ARG 2 component
```
Comment (sciwhiz12) [inline data/net/minecraft/server/packs/repository/Pack.mapping]: This is likely to be the `title` of the pack.

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]
Looks good apart from three parameter changes for the pack's `title`.

**Review by sciwhiz12 — APPROVED** [review]

**Review by MelanX — APPROVED** [review]

**Review by Lanse505 — APPROVED** [review]

**Pull request merged**